ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,214 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
+import { I18nBehavior } from '../nuxeo-i18n-behavior.js';
+import '@nuxeo/nuxeo-elements/nuxeo-element.js';
+import '../../core/nuxeo-page-provider.js';
+import '../search/nuxeo-search-form-layout.js';
+import '../search/nuxeo-search-results-layout.js';
+import '../widgets/nuxeo-dialog.js';
+import '@polymer/paper-button/paper-button.js';
+
+{
+  /**
+   * `nuxeo-document-picker`
+   * @group Nuxeo UI
+   * @element nuxeo-document-picker
+   */
+  class DocumentPicker extends mixinBehaviors([I18nBehavior], Nuxeo.Element) {
+    static get template() {
+      return html`
+        <style include="nuxeo-styles">
+          nuxeo-dialog {
+            display: flex;
+            flex-direction: column;
+            justify-content: space-between;
+            height: 100%;
+            width: 100%;
+            /* TODO check if we can use something a bit bigger here */
+            max-height: 84%;
+            max-width: 768px;
+            padding: 8px;
+            overflow: auto;
+          }
+          #topContainer {
+            display: flex;
+            justify-content: flex-end;
+          }
+          iron-icon {
+            opacity: 0.6;
+          }
+          iron-icon:hover {
+            cursor: pointer;
+            opacity: 1;
+          }
+          #mainContainer,
+          #topContainer,
+          nuxeo-results-view {
+            margin: 0;
+            padding: 0;
+          }
+          nuxeo-results-view {
+            --nuxeo-results-view-height: calc(100vh - 520px - var(--nuxeo-app-top));
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[params]]"
+              quick-filters="{{quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{aggregations}}"
+              visible
```

Comment:
🤦‍♂️  just realized we have an element in elements which is hidden by default 😢

Reply:
I saw that there was a Slack discussion around this. Is there a plan for this?

Reply:
[ELEMENTS-1331](https://jira.nuxeo.com/browse/ELEMENTS-1331): Allow nuxeo-results-view to be used without having to explicitly set it to visible